File: week-07/day-2/Workshop 16.5.2023/product_database.py
products = {}
products['Eggs'] = 200
products['Milk'] = 200
products['Fish'] = 400
products['Apples'] = 150
products['Bread'] = 50
products['Chicken'] = 550

fish = products.get('Fish')
print(fish)

most_expensive = ' '
cost = 0
for item in products:
    if products.get(item) > cost:
        cost = products.get(item)
        most_expensive = item

print(most_expensive)

average = 0
for item in products:
    average = average+products.get(item)

average = average/len(products)
print(average)

money = 125
if money in products.values():
    print('yes')
else:
    print('no')


cheapest = ''
cost = 10000
for item in products:
    if products.get(item) < cost:
        cost = products.get(item)
        cheapest = item

print(cheapest)

File: week-02/day-4/Workshop 13.4.2023/src/ProductDatabase.java
import java.util.HashMap;
import java.util.Map;

public class ProductDatabase {
    public static void main(String[] args) {
        HashMap<String, Integer> products = new HashMap<>();
        products.put("Eggs", 200);
        products.put("Milk", 200);
        products.put("Fish", 400);
        products.put("Apples", 150);
        products.put("Bread", 50);
        products.put("Chicken", 550);

        //How much is the fish?
        System.out.println(products.get("Fish"));

        //What is the most expensive product?
        int mostExpensive = 0;
        String key = "0";
        for (String prod : products.keySet()) {
            products.get(prod);
            if (mostExpensive <= products.get(prod)) {
                mostExpensive = products.get(prod);
                key = prod;
            }
        }
        System.out.println(key);

        //What is the average price?
        double averagePrice = 0;
        for (double price : products.values()) {
            averagePrice = averagePrice + price;
        }
        averagePrice = averagePrice / products.size();
        System.out.println(averagePrice);

       // How many products' price is below 300?
        int counter = 0;
        for (String priceBelow300 : products.keySet()){
           if  (products.get(priceBelow300) < 300) {
               counter ++;
           }
        }
        System.out.println(counter);

        // Is there anything we can buy for exactly 125?
        for (String price125 : products.keySet()) {
            if (products.get(price125) == 125) {
                System.out.println("yes");
            }
        }
        System.out.println("no");

        //What is the cheapest product?
        int cheapest = 500;
        String cheapestProduct = "0";
        for (String prod : products.keySet()) {
            products.get(prod);
            if (cheapest >= products.get(prod)) {
                cheapest = products.get(prod);
                cheapestProduct = prod;
            }
        }
        System.out.println(cheapestProduct);
    }
}
